.github/workflows/nightly.yml

```diff
@@ -94,3 +94,46 @@ jobs:
       s3_bucket: ${{ secrets.IT_CASE_S3_BUCKET }}
       s3_access_key: ${{ secrets.IT_CASE_S3_ACCESS_KEY }}
       s3_secret_key: ${{ secrets.IT_CASE_S3_SECRET_KEY }}
+
+  build_python_wheels:
+    name: "Build Python Wheels on ${{ matrix.os_name }}"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            os_name: linux
+          - os: macos-latest
+            os_name: macos
+    steps:
+      - name: "Checkout the repository"
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+      - name: "Stringify workflow name"
+        uses: "./.github/actions/stringify"
+        id: stringify_workflow
+        with:
+          value: ${{ github.workflow }}
+      - name: "Build python wheels for ${{ matrix.os_name }}"
+        uses: pypa/cibuildwheel@v2.16.5
```

Comment:
Interesting approach. It's just that Apache doesn't allow actions other than the ones from the `apache`, `github` and `actions` ([source](https://infra.apache.org/github-actions-policy.html)). We could use a fixed version for the custom action and review that specific version. But `pypa/cibuildwheel@v2.16.5` seems to be a bigger project. I'm wondering whether it's worth it or just use the previous approach (even though I liked your intention 👍 ). WDYT?

Reply:
Ahh I see. I would be fine to use generic install aproach, right?

```
      # Used to host cibuildwheel
      - uses: actions/setup-python@v5

      - name: Install cibuildwheel
        run: python -m pip install cibuildwheel==2.18.1

      - name: Build wheels
        run: python -m cibuildwheel --output-dir wheelhouse
```

cibuildwheel docs also suggest this generic way for GitHub actions

Reply:
That should work 👍

Reply:
Great, updated ✔️

Reply:
Looks fine from my side. But I am not familiar with the whole buildwheel logic. @HuangXingBo can you do another pass over it and approve the changes once more?